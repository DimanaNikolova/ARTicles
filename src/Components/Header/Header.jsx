import React, { Component } from 'react'
import './Header.css'
import UserContext from '../../utils/UserContext'
import MenuGroup from '../MenuGroup'

export default class Header extends Component {
    static contextType = UserContext
    render() {
        const { user } = this.context
        return <nav className="menu">
        {user ?
            <ul className="menu__list">
                <MenuGroup path='/' name='Home' />
                <MenuGroup path='/create-article' name='Create' />
                <MenuGroup path={`/profile/${this.context.user.userId}`} name='Profile' />
                <MenuGroup path='/' name='Logout' onClick={this.context.logOut}/>
            </ul>
            :
            <ul className="menu__list">
                <MenuGroup path='/' name='Home' />
                <MenuGroup path='/register' name='Register' />
                <MenuGroup path='/login' name='Login' />
            </ul>}
    </nav>
    }
}
