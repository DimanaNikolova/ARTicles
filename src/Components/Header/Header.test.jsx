import React from 'react'
import renderer from 'react-test-renderer'
import Header from './Header'
import TestingEnvironment from '../../utils/router-test'

describe('Header Component', () => {
  it('should render authenticated routes', () => {
    const tree = renderer.create(
      <TestingEnvironment value={{
        user: true
      }}>
        <Header />
      </TestingEnvironment>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('render unauthenticated routes', () => {
    const tree = renderer.create(
      <TestingEnvironment value={{
        user: false
      }}>
        <Header />
      </TestingEnvironment>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})