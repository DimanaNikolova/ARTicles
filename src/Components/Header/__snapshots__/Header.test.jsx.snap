// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component render unauthenticated routes 1`] = `
<nav
  className="menu"
>
  <ul
    className="menu__list"
  >
    <li
      className="menu__group"
    >
      <a
        className="menu__link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="menu__group"
    >
      <a
        className="menu__link"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </li>
    <li
      className="menu__group"
    >
      <a
        className="menu__link"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Header Component should render authenticated routes 1`] = `
<nav
  className="menu"
>
  <ul
    className="menu__list"
  >
    <li
      className="menu__group"
    >
      <a
        className="menu__link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="menu__group"
    >
      <a
        className="menu__link"
        href="/create-article"
        onClick={[Function]}
      >
        Create
      </a>
    </li>
    <li
      className="menu__group"
    >
      <a
        className="menu__link"
        href="/profile/undefined"
        onClick={[Function]}
      >
        Profile
      </a>
    </li>
    <li
      className="menu__group"
    >
      <a
        className="menu__link"
        href="/"
        onClick={[Function]}
      >
        Logout
      </a>
    </li>
  </ul>
</nav>
`;
